### DISCUSSION TITLE
Secure Payment Request

### DISCUSSION THREAD

**Discussion by Nevercold** [web-auth/webauthn-framework#506] [Ideas]
Where I just browsed "SimpleWebAuthn", I saw the possibility of "Secure Payment Request".

Is this already supported by this library? Or is this planned?

https://www.w3.org/TR/secure-payment-confirmation/
https://github.com/MasterKale/SimpleWebAuthn/discussions/402

**Comment by Spomky** [comment]
Hi @Nevercold,

I was not aware of this specification. Many thanks for the hint! 🙂

I read this page from Matthew Miller (https://simplewebauthn.dev/docs/advanced/secure-payment-confirmation) and it does not seem to be difficult to have it here.
If I understand clearly, the library should support a new `payment.get` type instead of just `webauthn.get`. Types could have their own logic and, instead of [throwing an exception here](https://github.com/web-auth/webauthn-framework/blob/4.7.x/src/webauthn/src/AuthenticatorAssertionResponseValidator.php#L194), the supported type should be called.
I am working on the branch 4.8.x/5.0.x for early 2024. I will try to introduce something for supporting it in these branches.
